### PR TITLE
Fix Akamai duplicated & missing events issue

### PR DESCRIPTION
**Pull request by YuvHayun**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py [hunk 1]
```diff
@@ -96,14 +96,16 @@ def get_events_with_offset(
         self,
         config_ids: str,
         offset: str | None = '',
-        limit: str | int | None = None,
-        from_epoch: str | None = ''
+        limit: int = 20,
```
Comment (JudahSchwartz) [inline Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:99]: this isnt a bw change is it?

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py [hunk 2]
```diff
@@ -169,7 +171,7 @@ def decode_message(msg: str) -> Sequence[str | None]:
     readable_msg = []
     translated_msg = urllib.parse.unquote(msg).split(';')
     for word in translated_msg:
-        word = b64decode(word.encode('utf8')).decode('utf8')
+        word = b64decode(word).decode('utf-8', errors='replace')
```
Comment (JudahSchwartz) [inline Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:174]: whats this do? we arent expecting it in as bytes anymore?

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py [hunk 3]
```diff
@@ -394,34 +396,37 @@ def fetch_events_command(
...
+                                            'ruleActions', 'ruleVersions']:
+                        event['attackData'][attack_data_key] = decode_message(event.get('attackData', {}).get(attack_data_key,
+                                                                                                              ""))
                 if "httpMessage" in event:
                     event['httpMessage']['requestHeaders'] = decode_url(event.get('httpMessage', {}).get('requestHeaders', ""))
                     event['httpMessage']['responseHeaders'] = decode_url(event.get('httpMessage', {}).get('responseHeaders', ""))
             except Exception as e:
                 config_id = event.get('attackData', {}).get('configId', "")
                 policy_id = event.get('attackData', {}).get('policyId', "")
                 demisto.debug(f"Couldn't decode event with {config_id=} and {policy_id=}, reason: {e}")
```
Comment (JudahSchwartz) [inline Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:424]: why only debug level?

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py (hunk 1 above)
Comment (YuvHayun) [reply Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:99]: Nope.

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py (hunk 2 above)
Comment (YuvHayun) [reply Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:174]: Fixing decoding issues we had, me & @dantavori looked at it together yesterday.

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py (hunk 3 above)
Comment (YuvHayun) [reply Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:424]: prefer info? Idm.

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py [hunk 4]
```diff
@@ -394,34 +396,37 @@ def fetch_events_command(
         (list[dict], str): events and new offset.
```
Comment (dantavori) [inline Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:396]: please update docstring

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py [hunk 5]
```diff
@@ -478,16 +483,17 @@ def main():
...
                 params.get("configIds"),
                 ctx=get_integration_context() or {},
             ):
-                send_events_to_xsiam(events, VENDOR, PRODUCT, should_update_health_module=False)
-                set_integration_context({"offset": offset})
-            demisto.updateModuleHealth({'eventsPulled': total_events_count})
+                if events:
+                    send_events_to_xsiam(events, VENDOR, PRODUCT, should_update_health_module=False)
+                set_integration_context({"offset": offset, "last_run_time": new_from_time})
+            demisto.updateModuleHealth({'eventsPulled': (total_events_count or 0)})
```
Comment (dantavori) [inline Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:496]: why is the "or 0" necessary?

File: Packs/Akamai_SIEM/ReleaseNotes/1_1_4.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+
+##### Akamai WAF SIEM
+
+- Fixed an issue where **fetch-events** would sometimes miss / duplicate events due to discrepancy in the fetching mechanism.
+- Fixed an issue where fetched events will sometimes fail decoding certain fields.
```
Comment (dantavori) [inline Packs/Akamai_SIEM/ReleaseNotes/1_1_4.md:7]:
```suggestion
- Fixed an issue where the ***fetch-events*** command would sometimes miss or duplicate events.
- Fixed an issue where fetched events sometimes failed to decode certain fields.
```

**Review by dantavori — APPROVED** [review]
nice, see notes

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py (hunk 1 above)
Comment (dantavori) [reply Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:99]: @YuvHayun do we intend to keep this?

File: Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py (hunk 5 above)
Comment (YuvHayun) [reply Packs/Akamai_SIEM/Integrations/Akamai_SIEM/Akamai_SIEM.py:496]: to cover cases where total_events_count is not defined (happened according to the logs).

**Comment by YuvHayun** [comment]
Need force merge due to [known reason with saas tpb](https://jira-dc.paloaltonetworks.com/browse/CRTX-87440) @dantavori

**Pull request merged**